libtransmission/torrent-magnet.h

```diff
@@ -23,36 +25,62 @@ struct tr_torrent;
 struct tr_torrent_metainfo;
 
 // defined by BEP #9
-inline constexpr int MetadataPieceSize = 1024 * 16;
+inline constexpr auto MetadataPieceSize = 1024 * 16;
 
 using tr_metadata_piece = small::max_size_vector<std::byte, MetadataPieceSize>;
 
-struct tr_incomplete_metadata
+class tr_metadata_download
 {
-    struct metadata_node
+public:
+    struct Mediator
     {
-        time_t requested_at = 0U;
-        int piece = 0;
+        virtual ~Mediator() = default;
+
+        [[nodiscard]] virtual std::string log_name() const noexcept = 0;
     };
```

Comment:
If we only need the mediator for one thing, can we pass the logname into the `tr_metadata_download()` constructor and remove the mediator?

Reply:
1dd8fe3